### PR TITLE
New JSON Validator

### LINKED ISSUE #3431: Suggestion: Add JSON Validator
_(Bringing this from forum as an issue)_

I will work on this. I have found this [pretty neat snippet](http://stackoverflow.com/a/12217209) to check if the json is valid or not. Please correct me, if I am wrong with this one.

EDIT: [Alternative solution](http://stackoverflow.com/a/20392392) (Without using jQuery)

/cc @MariagraziaAlastra @moollaza

### PR DESCRIPTION
**Pull request by sahildua2305**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
    - [x] Other: **`New JSON Validator Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Which issues (if any) does this fix?

Fixes #3431 

###### People to notify (@mention interested parties)
@MariagraziaAlastra @moollaza 

---

Instant Answer Page: https://duck.co/ia/view/json_validator
Forum topic: [Create IA: JSON Lint](https://forum.duckduckhack.com/t/create-ia-json-lint/327)

This pull request is part of the [Programming Mission](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53): help us improve the results for [JavaScript related searches](https://forum.duckduckhack.com/t/javascript-search-overview/94)!

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by edgesince84** [comment]
👍

**Comment by MariagraziaAlastra** [comment]
Hey @sahildua2305, it's deployed on [beta](https://beta.duckduckgo.com/?q=json+validator&ia=answer)!
screenshot-beta duckduckgo com 2016-07-25 14-54-34

Can you please disable the button completely until the lib is loaded? I can still click it, and it will output the error shown above. 
Thanks!

**Comment by sahildua2305** [comment]
Fixed, @MariagraziaAlastra! There was a minor typo in the template.

**Comment by MariagraziaAlastra** [comment]
@sahildua2305 Thanks, this looks good! I'm adding the library internally, will let you know when it's available on beta to test :smile:

File: share/goodie/jsonvalidator/jsonvalidator.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                    .css('cursor', 'pointer')
+                    .removeClass('btn--skeleton')
+                    .addClass('btn--primary');
+            });
+
+            $validateButton.click(function () {
+                $result.parent().removeClass('is-hidden');
+                try {
+                    var result = jsonlint.parse($input.val());
+                    if (result) {
```
Comment (MariagraziaAlastra) [inline share/goodie/jsonvalidator/jsonvalidator.js]: @sahildua2305 Could you also prettyprint the JSON if it's valid? 
"jsonlint will either report a syntax error with details or pretty print the source if it is valid." that's what it says in the library's [readme](https://github.com/zaach/jsonlint/blob/master/README.md), although I didn't investigate it.

If it's easy to do it'd be a way to integrate a visualizer here and we could expand the triggers as well!

Comment (sahildua2305) [inline share/goodie/jsonvalidator/jsonvalidator.js]: As we discussed on Slack, I will integrate that feature once basic JSON Validator is live. :smile:

**Comment by MariagraziaAlastra** [comment]
@sahildua2305 It's on [beta](https://beta.duckduckgo.com/?q=json+validator&ia=answer)! 
Looks good to me, merging :+1: 
Thanks, great job here! Next step is prettyprinting the JSON if it's valid :smile: 

This should be live by Wednesday next week!

**Comment by daxtheduck** [comment]
@sahildua2305 this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by sahildua2305** [comment]
Thanks @MariagraziaAlastra 

I will push that feature over the weekend. :smile:

**Pull request merged**